### PR TITLE
feat: support postgresql16

### PR DESCRIPTION
**Pull request by Y-Rookie**
refer: https://github.com/apecloud/kubeblocks/issues/7994

### CODE REVIEW

File: addons/postgresql/templates/clusterversion.yaml [hunk 1]
```diff
@@ -245,4 +245,50 @@ spec:
...
+  labels:
+   {{- include "postgresql.labels" . | nindent 4 }}
+spec:
+  clusterDefinitionRef: postgresql
+  componentVersions:
+    - componentDefRef: postgresql
+      configSpecs:
+        - name: postgresql-configuration
+          templateRef: postgresql16-configuration
+          constraintRef: postgresql15-cc
```
Comment (nayutah) [inline addons/postgresql/templates/clusterversion.yaml:268]: pg16 use pg15 cc ?

File: addons/postgresql/templates/componentdefinition-16.yaml [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+    scrapePath: /metrics
+  configs:
+    - name: postgresql-configuration
+      templateRef: postgresql16-configuration
+      constraintRef: postgresql15-cc
+      keys:
+        - postgresql.conf
+      namespace: {{ .Release.Namespace }}
+      volumeName: postgresql-config
+      defaultMode: 0444
```
Comment (nayutah) [inline addons/postgresql/templates/componentdefinition-16.yaml:53]: does 0444 support a online reconfigure ?

File: addons/postgresql/templates/componentdefinition-16.yaml [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+      withCandidate:
+        image: {{ .Values.image.registry | default "docker.io" }}/{{ .Values.image.repository }}:15.7.0
+        exec:
+          command:
+            - /bin/bash
+            - -c
+          args:
+            - curl -s http://$(KB_REPLICATION_PRIMARY_POD_FQDN):8008/switchover -XPOST -d '{"leader":"$(KB_REPLICATION_PRIMARY_POD_NAME)","candidate":"$(KB_SWITCHOVER_CANDIDATE_NAME)"}'
+      withoutCandidate:
+        image: {{ .Values.image.registry | default "docker.io" }}/{{ .Values.image.repository }}:15.7.0
```
Comment (nayutah) [inline addons/postgresql/templates/componentdefinition-16.yaml:114]: 15.7 for pg16?

File: addons/postgresql/templates/componentdefinition-16.yaml [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+              fieldRef:
+                fieldPath: metadata.annotations['rs.apps.kubeblocks.io/primary']
+            - path: "component-replicas"
+              fieldRef:
+                fieldPath: metadata.annotations['apps.kubeblocks.io/component-replicas']
+  vars:
+    - name: POSTGRES_USER
+      valueFrom:
+        credentialVarRef:
+          compDef: postgresql-15
```
Comment (nayutah) [inline addons/postgresql/templates/componentdefinition-16.yaml:337]: ditto

File: addons/postgresql/templates/clusterversion.yaml (hunk 1 above)
Comment (Y-Rookie) [reply addons/postgresql/templates/clusterversion.yaml]: > pg16 use pg15 cc ?

fixed

File: addons/postgresql/templates/componentdefinition-16.yaml (hunk 2 above)
Comment (Y-Rookie) [reply addons/postgresql/templates/componentdefinition-16.yaml]: > does 0444 support a online reconfigure ?

fixed

**Review by nayutah — APPROVED** [review]

**Pull request merged**